test/factory/dart_function_factory_test.dart

```diff
@@ -0,0 +1,30 @@
+// ignore_for_file: deprecated_member_use_from_same_package
+// ignore_for_file: invalid_use_of_protected_member
+@TestOn('browser')
+
+import 'package:test/test.dart';
+import 'package:react/react.dart' as react;
+import 'package:react/react_client.dart';
+import 'common_factory_tests.dart';
+
+main() {
+  setClientConfiguration();
+
+  group('ReactDartFunctionComponentFactoryProxy', () {
+    test('has a type corresponding to the backing JS Function', () {
+      expect(FunctionFoo.type, equals(FunctionFoo.reactFunction));
+    });
+
+    group('Function Component -', () {
+      group('- common factory behavior -', () {
+        commonFactoryTests(FunctionFoo, isFunctionComponent: true);
+      });
+    });
+  });
+}
+
+final FunctionFoo = react.registerFunctionComponent(_FunctionFoo);
+
+_FunctionFoo(Map props) {
+  return react.div({});
```

Comment:
Do the `commonFactoryTests` not test any sort of prop forwarding? Seems like the `props` value is completely ignored here - which I would expect to cause at least 1 failing test.

Reply:
yeah they dont, this is basicly copypasta from `dart_factory_test.dart`
```
final Foo = react.registerComponent(() => new _Foo()) as ReactDartComponentFactoryProxy;

class _Foo extends react.Component {
  @override
  render() => react.div({});
}

final Foo2 = react.registerComponent(() => new _Foo2()) as ReactDartComponentFactoryProxy2;

class _Foo2 extends react.Component2 {
  @override
  render() => react.div({});
}
```

the props are tested pretty exhaustively in `lifecycle_test.dart` though